src/server/login/login-middleware.js

```diff
@@ -74,34 +98,43 @@ const setup = (app: Router) => {
   app.post(
     '/auth/eth',
     wrapAsync(async (req, res) => {
-      const log = req.log
+      const { log, body } = req
+      const { network_id: configNetworkId } = Config.ethereum
 
       log.debug('/auth/eth', { message: 'authorizing' })
-      log.debug('/auth/eth', { body: req.body })
+      log.debug('/auth/eth', { body })
+
+      const {
+        nonce,
+        method,
+        networkId,
+        signature,
+        gdSignature,
+        profileSignature,
+        profilePublickey: profileReqPublickey
+      } = body
 
-      const signature = req.body.signature
-      const gdSignature = req.body.gdSignature
-      const profileReqPublickey = req.body.profilePublickey
-      const profileSignature = req.body.profileSignature
-      const nonce = req.body.nonce
-      const method = req.body.method
-      const networkId = req.body.networkId
+      log.debug('/auth/eth', { signature, method })
 
-      if (networkId !== Config.ethereum.network_id) {
+      if (networkId !== configNetworkId) {
         log.warn('/auth/eth', {
           message: 'Networkd id mismatch',
           client: networkId,
-          server: Config.ethereum.network_id
+          server: configNetworkId
         })
-        throw new Error(`Network ID mismatch client: ${networkId} ours: ${Config.ethereum.network_id}`)
+
+        throw new Error(`Network ID mismatch client: ${networkId} ours: ${configNetworkId}`)
       }
-      log.debug('/auth/eth', { signature, method })
+
+      isProfileSignatureCompatible(profileSignature, nonce)
```

Comment:
why are we checking it twice?
we are not catching excption

Reply:
This occurrence should be removed